pkg/policy/compiler.go

```diff
@@ -83,23 +88,44 @@ func (c *compiler) Compile(policy *v1alpha1.AuthorizationPolicy) (PolicyFunc, fi
 			variables[variable.Name] = prog
 		}
 	}
-	var authorizations []cel.Program
+	var authorizations []AuthorizationPrograms
 	{
 		path := path.Child("authorizations")
 		for i, rule := range policy.Spec.Authorizations {
 			path := path.Index(i)
-			ast, issues := env.Compile(rule.Expression)
-			if err := issues.Err(); err != nil {
-				return nil, append(allErrs, field.Invalid(path.Child("expression"), rule.Expression, err.Error()))
-			}
-			if !ast.OutputType().IsExactType(envoy.CheckResponse) {
-				return nil, append(allErrs, field.Invalid(path.Child("expression"), rule.Expression, "rule output is expected to be of type envoy.service.auth.v3.CheckResponse"))
+			program := AuthorizationPrograms{}
+
+			{
+				ast, issues := env.Compile(rule.Response)
+				if err := issues.Err(); err != nil {
+					return nil, append(allErrs, field.Invalid(path.Child("response"), rule.Response, err.Error()))
+				}
+				if !ast.OutputType().IsExactType(envoy.CheckResponse) {
+					return nil, append(allErrs, field.Invalid(path.Child("response"), rule.Response, "rule response output is expected to be of type envoy.service.auth.v3.CheckResponse"))
+				}
+				prog, err := env.Program(ast)
+				if err != nil {
+					return nil, append(allErrs, field.Invalid(path.Child("response"), rule.Response, err.Error()))
+				}
+				program.Response = prog
 			}
-			prog, err := env.Program(ast)
-			if err != nil {
-				return nil, append(allErrs, field.Invalid(path.Child("expression"), rule.Expression, err.Error()))
+
+			if rule.Match != "" {
```

Comment:
You should compile match first.

Reply:
donÄt get this point. Why should I compile it before I am checking if there is a match condition declared?